tests/recognizer-tests.js

```diff
@@ -129,6 +129,16 @@ test("A dynamic route recognizes", function() {
   equal(router.recognize("/zoo/baz"), null);
 });
 
+test("A dynamic route with encoded params", function() {
+  var handler = {};
+  var router = new RouteRecognizer();
+  router.add([{ path: "/foo/:bar", handler: handler }]);
+
+  equal(router.recognize("/foo/bar%20baz")[0].params.bar, "bar baz");
+  equal(router.recognize("/foo/bar%23baz")[0].params.bar, "bar#baz");
+  equal(router.recognize("/foo/bar%3fbaz")[0].params.bar, "bar?baz");
+});
+
```

Comment:
I think that the path segments should absolutely be URL decoded. I think that is equivalent to decoding the entire URL (I'm not sure the exact algorithm, but I think the decoding only affects sequences starting with a %?).

This would place nicely with future features like optional segments: `/lol(/%28wat%29)`. Heaven forbid someone actually want to do that.